Sources/SwiftInspectorVisitors/TypeDescription.swift

```diff
@@ -0,0 +1,267 @@
+      self = .optional(typeDescription)
+
+    } else if caseDescription == Self.implicitlyUnwrappedOptionalDescription {
+      let typeDescription = try values.decode(Self.self, forKey: .typeDescription)
+      self = .implicitlyUnwrappedOptional(typeDescription)
+
+    } else if caseDescription == Self.arrayDescription {
+      let typeDescription = try values.decode(Self.self, forKey: .typeDescription)
+      self = .array(typeDescription)
+
+    } else if caseDescription == Self.dictionaryDescription {
+      let key = try values.decode(Self.self, forKey: .typeDescriptionDictionaryKey)
+      let value = try values.decode(Self.self, forKey: .typeDescriptionDictionaryValue)
+      self = .dictionary(key: key, value: value)
+
+    } else if caseDescription == Self.compositionDescription {
+      let typeDescriptions = try values.decode([Self].self, forKey: .typeDescriptions)
+      self = .composition(typeDescriptions)
+
+    } else if caseDescription == Self.tupleDescription {
+      let typeDescriptions = try values.decode([Self].self, forKey: .typeDescriptions)
+      self = .tuple(typeDescriptions)
+
+    } else {
+      throw CodingError.unknownCase
+    }
+  }
+
+  public func encode(to encoder: Encoder) throws {
+    var container = encoder.container(keyedBy: CodingKeys.self)
+    try container.encode(caseDescription, forKey: .caseDescription)
+    switch self {
+    case let .simple(name):
+      try container.encode(name, forKey: .text)
+    case let .unknown(text):
+      try container.encode(text, forKey: .text)
+    case let .optional(type),
+         let .array(type),
+         let .implicitlyUnwrappedOptional(type):
+      try container.encode(type, forKey: .typeDescription)
+    case let .tuple(types),
+         let .composition(types):
+      try container.encode(types, forKey: .typeDescriptions)
+    case let .dictionary(key, value):
+      try container.encode(key, forKey: .typeDescriptionDictionaryKey)
+      try container.encode(value, forKey: .typeDescriptionDictionaryValue)
+    case let .nested(name, parentType):
+      try container.encode(name, forKey: .text)
+      try container.encode(parentType, forKey: .typeDescription)
+    }
+  }
+
+  enum CodingKeys: String, CodingKey {
+    /// The value for this key is the case encoded as a string.
+    case caseDescription
+    /// The value for this key is an associated value of type String
+    case text
+    /// The value for this key is the associated value of type TypeDescription
+    case typeDescription
+    /// The value for this key is the associated value of type [TypeDescription]
+    case typeDescriptions
+    /// The value for this key is the associated value of type TypeDescription that represents the key in a dictionary.
+    case typeDescriptionDictionaryKey
+    /// The value for this key is the associated value of type TypeDescription that represents the value in a dictionary
+    case typeDescriptionDictionaryValue
+  }
+
+  enum CodingError: Error {
```

Comment:
Should we make this public so someone could identify it in their consuming code?

Reply:
9c375bad2a837b3f8a1fa027828df3b68c84523f